### DISCUSSION TITLE
Can I install visual-studio-code via brew on a arm (Apple Silicon) arch?

### DISCUSSION THREAD

**Discussion by nbro** [Homebrew/discussions#5446] [Getting started]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.6
ORIGIN: https://github.com/Homebrew/brew
HEAD: e8430b25a1d1321f32e5093d62b57b5cb7cfb3c3
Last commit: 2 days ago
Core tap JSON: 19 Jun 02:17 UTC
Core cask tap JSON: 19 Jun 02:28 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.3.3 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.3/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.6.0 => /usr/bin/curl
macOS: 14.5-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

The reason why I'm asking this is that https://formulae.brew.sh/cask/visual-studio-code only shows variations for Intel, but I was able to install VS Code using brew and it seems to work. The cask code https://github.com/Homebrew/homebrew-cask/blob/a2004664a0017ea485d70ef2868d43b80f23118f/Casks/v/visual-studio-code.rb seems to have some part related to arm, so maybe the documentation is outdated?

In general, how do I know if a cask or a regular package is compatible with my arch when using brew?

**Comment by jloveland** [comment]
same problem here

**Comment by p-linnane** [comment]
It should have installed the arm64 version because your prefix is `/opt/homebrew` and you aren't using Rosetta. This is most likely a bug with the site rendering.

**Reply by KishanPipariya** [reply]
The site currently shows both Apple Silicon and Intel. So it appears that the bug was resolved at some point.

**Comment by blakewilson** [comment]
Running into the same thing.

**Reply by jloveland** [reply]
I solved this by installing the arm version of homebrew, fixing the path to point to the new /opt/homebrew location and ensuring my ZSH and environment recognized I was running arm instead of intel

**Reply by blakewilson** [reply]
That did it. Appreciate it!
